### PR TITLE
Add GLES dependency graph unit test template, and glClear() dependencies

### PR DESCRIPTION
**Pull request by hevrard**
Also introduce the template for dependency graph "unit tests"

### CODE REVIEW

**Comment by hevrard** [comment]
I'm mostly interested in the feedback on `dependencygraph2_test.go`. The idea is to have unit tests which build a small capture, get its dependency graph, and check that expected dependencies are indeed present.

Let me know if the current approach looks good or not, and how to improve it.

**Comment by pmuetschard** [comment]
looks good to me.

**Comment by hevrard** [comment]
Updated with dependencies for `glClear{Color,Depthf,Stencil}`

File: gapis/api/gles/dependencygraph2_test.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+	for dep, present := range mapToCheck {
+		assert.For(tc.ctx, fmt.Sprintf("Dependency: %v", dep)).ThatBoolean(present).IsTrue()
+	}
+}
+
+// Actual tests
+
+// glClear(GL_COLOR_BUFFER_BIT) depends on glClearColor()
+func TestDependencyGlClearColor(t *testing.T) {
```
Comment (AWoloszyn) [inline gapis/api/gles/dependencygraph2_test.go]: One comment I would like to make is that a dependency graph that just returned "KEEP EVERYTHING" would also pass these tests. 

If we want to write unit tests for the dependency graph, we should also at least make sure that some of the edges are NOT present.

e.g. 
glClear(GL_COLOR_BUFFER_BIT) does not depend on glClearDepthf()

Obviously we can't be exhaustive, but we should at least attempt in some cases to make sure it is not a complete graph.

File: gapis/api/gles/dependencygraph2_test.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	"github.com/google/gapid/gapis/memory"
+	"github.com/google/gapid/gapis/resolve/dependencygraph2"
+)
+
+// Test helpers
+
+// testDependencyCmdName is used as a map key to easily express
+// expected dependencies
+type testDependencyCmdName struct {
+	sourceCmdName, targetCmdName string
```
Comment (bjoeris) [inline gapis/api/gles/dependencygraph2_test.go]: It seems that identifying nodes based only on the command name will only work for very small tests (with just 1 call to each distinct command). But it does work for the tests here.

Comment (hevrard) [reply gapis/api/gles/dependencygraph2_test.go]: Yes I also though about this, I feel we may have to enrich the way we identify a node later on. I feel like I need to face more complex captures to know what is best to do, so I'll leave this for a future change.

File: gapis/api/gles/dependencygraph2_test.go (hunk 1 above)
Comment (hevrard) [reply gapis/api/gles/dependencygraph2_test.go]: Very good point, I updated the unit test template to take two list of dependencies: expected and unexpected dependencies.

**Pull request merged**